### PR TITLE
Adding check if dependencies key is present

### PR DESCRIPTION
**Pull request by maitrayshah**
NPMPackageVersionScanner

- Adding a check to validate `dependencies` are present in `package-lock.json`.
- Adding spec to handle the above use case.

NPMAudit

- Changing `should_run?` condition to include both `package.json` and `package-lock.json`
- Updating incorrect test case to handle above use case.

### CODE REVIEW

**Comment by saikat056** [comment]
Can we add a unit test for this change? Thank you.

File: lib/salus/package_utils/package_lock_json.rb [hunk 1]
```diff
@@ -56,9 +56,11 @@ def record_dep_locations
 
     # recursively store all dependency names as keys
     def get_dep_names(data)
-      data['dependencies'].each do |name, dep_info|
-        @deps[name] = {}
-        get_dep_names(dep_info) if dep_info['dependencies']
+      if data.key?("dependencies")
```
Comment (joshua-ostrom) [inline lib/salus/package_utils/package_lock_json.rb:59]: return unless data.key?("dependencies")
might be closer to the ruby style guide 


Bigger ask though is can we add a spec for the behavior?

Comment (maitrayshah) [reply lib/salus/package_utils/package_lock_json.rb:59]: Yup adding

File: lib/salus/package_utils/package_lock_json.rb [hunk 2]
```diff
@@ -56,6 +56,8 @@ def record_dep_locations
 
     # recursively store all dependency names as keys
     def get_dep_names(data)
+      return unless data.key?("dependencies")
```
Comment (saikat056) [inline lib/salus/package_utils/package_lock_json.rb:59]: It would be great if we can have a unit test related the changes in 'def get_dep_names'.

We can add the unit test in a separate pr.

It's not recommended to use get_* or set_* at the beginning of method name in ruby? Can we use a method name like 'def generate_dep_names_for(data)'.

We can make these changes in a separate pr.

**Review by saikat056 — APPROVED** [review]
